baseapp/abci.go

```diff
@@ -780,6 +780,11 @@ func (app *BaseApp) internalFinalizeBlock(ctx context.Context, req *abci.Request
 
 	events = append(events, beginBlock.Events...)
 
+	// Reset the gas meter so that the AnteHandlers aren't required to
+	gasMeter = app.getBlockGasMeter(app.finalizeBlockState.ctx)
+	app.finalizeBlockState.ctx = app.finalizeBlockState.ctx.WithBlockGasMeter(gasMeter)
+	app.checkState.ctx = app.checkState.ctx.WithBlockGasMeter(gasMeter)
```

Comment:
sorry, why are we setting CheckTx state here?

Reply:
Good call out, there's no need to modify the checkTx state's meter here.